test/matrix_spec_coverage.py

```diff
@@ -0,0 +1,94 @@
+import sys
+import re
+import yaml
+from responses import RequestsMock
+
+INTERPOLATIONS = [
+    ("%CLIENT_MAJOR_VERSION%", "r0")
+]
+
+def interpolate_str(s):
+    for interpolation in INTERPOLATIONS:
+        s = s.replace(interpolation[0], interpolation[1])
+    return s
+
+def endpoint_to_regex(s):
+    # TODO sub by with more specific REGEXes per type
+    # e.g. roomId, eventId, userId
+    return re.sub('\{[a-zA-Z]+\}', '[a-zA-Z!\.:-@#]+', s)
+
+MISSING_BASE_PATH = "Not a valid API Base Path: "
+MISSING_ENDPOINT = "Not a valid API Endpoint: "
+MISSING_METHOD = "Not a valid API Method: "
+
+class ApiGuide:
```

Comment:
Not a fan of this class-name - but I'd prefer not to have something very long like `MatrixApiCoverageStatGatherer`, though maybe that's better than this?